### PR TITLE
Text Explainer Ops

### PR DESCRIPTION
**Pull request by boba-and-beer**
Need to add documentation on how to use the explainer into operators

### CODE REVIEW

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+
+class BaseExplainer(DocUtils):
+    """
+    Base class for vector
+    """
+
+    __name__ = ""
+
+    @classmethod
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:18]: why is this a `classmethod` ?

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                The list of fields to be used
+            field_type:
+                Accepts "vector" or "chunkvector"
+        """
+        for f in fields:
+            # Replace with case-switch in future
+            [
+                self._encode_document(
+                    encode_fn,
+                    f,
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:115]: more detailed variable names

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            missing_treatment:
+                Missing treatment can be one of ["do_not_include", "zero_vector", value].
+            documents:
+                The documents that are being used
+            fields:
+                The list of fields to be used
+            field_type:
+                Accepts "vector" or "chunkvector"
+        """
+        for f in fields:
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:110]: same here

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            [
+                self._encode_document(
+                    encode_fn,
+                    f,
+                    d,
+                    vector_error_treatment=vector_error_treatment,
+                    field_type=field_type,
+                )
+                for d in documents
+                if self.is_field(f, d)
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:121]: and here

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+                    doc=d,
+                    vector_error_treatment=vector_error_treatment,
+                    field_type="chunkvector",
+                )
+                for d in documents
+                if self.is_field(chunk_field, d)
+            ]
+        return documents
+
+    def encode_documents_in_bulk(
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:159]: `bulk_encode_documents`

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                d["_search_score"] = self.get_cosine_similarity(
+                    d[f"text_{self.__name__}_vector_"], query_vector
+                )
+            doc["explain_chunk"] = sorted(
+                doc["explain_chunk"], key=lambda x: x["_search_score"], reverse=True
+            )
+            cos_similarity.append(doc.copy())
+        return cos_similarity
+
+    def explain(
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:246]: this function is too similar to `explain_chunk` name wise. Documentation?

File: relevanceai/operations_new/cluster/text/explainer/base.py [hunk 7]
```diff
@@ -0,0 +1,268 @@
+"""Explainer
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/cluster/text/explainer/base.py:1]: where is `run` method ?

**Review by jtwinrelevanceai — CHANGES_REQUESTED** [review]
questions must be answered 🤔

File: relevanceai/operations_new/cluster/text/explainer/base.py (hunk 1 above)
Comment (boba-and-beer) [reply relevanceai/operations_new/cluster/text/explainer/base.py:18]: In case this is needed outside of this class.

**Comment by boba-and-beer** [comment]
Now part of SDK 390 2

**Pull request closed without merging**